packages/common/e2e/hub-project-class.e2e.ts

```diff
@@ -0,0 +1,104 @@
+import Artifactory from "./helpers/Artifactory";
+import config from "./helpers/config";
+import { Hub, HubProject } from "../src";
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 200000;
+
+function delay(milliseconds: number) {
+  return new Promise((resolve) => setTimeout(resolve, milliseconds));
+}
+
+fdescribe("HubProject Class", () => {
+  let factory: Artifactory;
+  beforeAll(() => {
+    jasmine.DEFAULT_TIMEOUT_INTERVAL = 200000;
+    factory = new Artifactory(config);
+  });
+  it("crud project", async () => {
+    // create context
+    const ctxMgr = await factory.getContextManager("hubBasic", "admin");
+    // create Hub
+    const myHub = await Hub.create({ contextManager: ctxMgr });
```

Comment:
Agreed on `.create` - perhaps `Hub.initialize(...)` for this? this is ~equivalent to the `GIS` in python, but seems it does not have explicit constructors (no `new`) and just takes args... we want to avoid `new` because that can't be async, so we use a factory fn.

As for the context/manager stuff... see the References and Immutable Objects part of this Guide -> https://esri.github.io/hub.js/guides/concepts/arcgiscontext/

Reply:
`initialize()` seems ok.

FWIW [Octokit](https://github.com/octokit/octokit.js/) uses new

```js
// Create a personal access token at https://github.com/settings/tokens/new?scopes=repo
const octokit = new Octokit({ auth: `personal-access-token123` });
```

it's curious to know what that does without async to validate.

Reply:
there are patterns that can do that deferred... `new UserSession(...)` does that in rest-js... but it does make debugging more O_o. The use of a factory function side-steps all that, and also means we _could_ change the implementation (which we can't do w/ `new`)